independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -33,7 +34,7 @@
     private static final Logger LOGGER = Logger.getLogger(RequestContext.class.getPackage().getName());
 
     // It's a normal scope so there may be no more than one mapped instance per contextual type per thread
-    private final ThreadLocal<ConcurrentMap<Contextual<?>, ContextInstanceHandle<?>>> currentContext = new ThreadLocal<>();
+    private final ThreadLocal<RequestContextState> currentContext = new ThreadLocal<>();
```

Comment:
Hm, this reminds me that we have an [open PR](https://github.com/quarkusio/quarkus/pull/22319) with conflicts... and  I'd like to merge it anyway. @manovotn @Ladicek your thoughts about https://github.com/quarkusio/quarkus/pull/22319 ?

Reply:
I am not against accepting #22319, but I don't think we should block this PR with it.

My 0.02$ are that #22319 only adds an abstraction that we need implementation for to make actual use of it. I can see Stuart made a [PoC](https://github.com/quarkusio/quarkus/pull/22319#issuecomment-998356103) that he asked @Sanne to review but I guess that got lost over xmass period. I'd be nice to pick up from there and once we see its viable, we can merge it as well.

Reply:
I'm very sorry about the delay on #22319 , I've had to set aside everything for a while and only returned back to work this week - Still catching up with things.

Reply:
Ok. I believe that the abstraction itself is worth it even without a further integration work but let's not block this PR (we'll resolve the conflicts later ;-).